app/src/main/java/pl/edu/zut/mad/appwizut2/connections/GetPlanChanges.java

```diff
@@ -0,0 +1,161 @@
+package pl.edu.zut.mad.appwizut2.connections;
+
+import android.util.Log;
+
+import org.apache.http.util.EntityUtils;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+
+import java.util.ArrayList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import pl.edu.zut.mad.appwizut2.models.MessagePlanChanges;
+
+
+/**
+ * Klasa do parsowania zmian w planie ze strony WI ZUT.
+ * 
+ * @author Sebastian Swierczek
+ * @version 1.2.1
+ */
+public class GetPlanChanges {
+
+	/**
+	 * Zmienna do debuggowania.
+	 */
+	private static final String TAG = "GetPlanChanges";
+
+	/** Zmienna przechowujaca zawartosc strony. */
+	private static String strona = "";
+
+	/**
+	 * Zmienna zawierajaca adres odnoszacy sie do zmian w planie w formacie JSON.
+	 * */
+	private static final String adres = "http://wi.zut.edu.pl/plan-zajec/zmiany-w-planie?format=json";
+
+	/** Zmienna zawierajaca szablon potrzebny przy parsowaniu */
+	private static final Pattern patternQuot = Pattern.compile("(&quot;)");
```

Comment:
Don't replace html entities when loading; keep them here as html and let's use [Html.fromHtml()](http://developer.android.com/reference/android/text/Html.html#fromHtml(java.lang.String)) when displaying

&amp;quot; is not only thing you might can encounter &lt;p&gt;, &lt;br&gt;, &amp;amp; are likely too.